main.js

```diff
@@ -97,8 +87,36 @@ class Timer {
         return this.started;
     }
 }
-const imgs = [];
-for(let i=0;i<12;i++){
-    imgs.push(document.getElementById('img'+i));
+
+class DOM_Manager {
+    constructor() {
+        this.cardsCountVar = document.getElementsByClassName("card").length;
+        this.imgs = [];
+        for(let i=0;i<this.cardsCountVar;i++){
+            this.imgs.push(document.getElementById("img"+i));
+        }
+    }
+    setListener(fun) {
+        for(let img of this.imgs) {
+            img.addEventListener('click', fun);

```

Comment:
You should call callback function with id if the card **only**.
Move this line here:
```js
let idClicked = parseInt(event.target.id.slice(3));
```

Reply:
@probil, please, give more info on this

Reply:
I mean something like this
```js
setListener(fun) {
        for(let img of this.imgs) {
            img.addEventListener('click', (event) => {  // `event` here is part of the DOM API so we need to keep it inside this class
              const idClicked = parseInt(event.target.id.slice(3));
              fn(idClicked);
           });
        }
}
```
After that consider moving callback to separate method because `removeEventListener` won't work without function reference

Reply:
@probil, but is there a reason to add a removeEventListener method? If card isn't visible - it isn't accesible. Or not?

Reply:
I think click will work even with `visibility: hidden`
But there is CSS property to prevent that
https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events

According to Data Driven Development your data/state should control how everything is rendered (including clicks, etc). It's easier to reason about as everything is controlled by flag somewhere. Your state is single source of truth. That's how modern FE frameworks work.

Currently you use simple data structure to describe cards like so:
```js
[1,4,1,3,5,4,...]; // <-- unique id of image to guess
```
Consider how you code and logic will change if your data structure will become something like this:
```js
const cards =  [
  {
    id: 1, // <-- unique id of the card
    hiddenValue: 3,  // <-- the id of image you try to guess
    isGuessed:  false,
    isOpened: false
  }, 
  {
    id: 2,
    hiddenValue: 2,
    isGuessed:  true, // <-- this card is hidden, ignores clicks etc
    isOpened: false
  },
  {
    id: 3,
    hiddenValue: 2,
    isGuessed:  true,
    isOpened: false
  },
  {
    id: 4,
    hiddenValue: 2,
    isGuessed:  false,
    isOpened: true, // <-- user just clicked on this card
  }, 
];
```
So it's much easier for `DOM_Manager` and `Game` to perform checks.
For example:
```js
// you don't have to update `nowOpened` all the time as it can be computed when needed (e.g. getter)
const nowOpened = cards.filter(card => card.isOpened).length;
```
It's really interesting topic!
If you are going to refactor you application this way - let's do it in the separate PR. Ok?
Let me know if you are interested

Reply:
`removeEventListener` is useful for preventing unexpected behavior when you re-using DOM elements. It's also kind of memory saving technique